### PR TITLE
wip: Replace libindy by vdr-tools

### PR DESCRIPTION
**Pull request by Patrik-Stas**
- Refers to my fork of vdr-tools with updated `libc`, otherwise `aries-vcx` doesn't compile (created PR https://gitlab.com/evernym/verity/vdr-tools/-/merge_requests/38 )
- Replace pgsql plugin test by mysql wallet test

Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    mod test {
+        use std::env;
+        use std::fs::File;
+        use std::io::Read;
+
+        use serde_json::Value;
+        use uuid::Uuid;
+
+        use aries_vcx::{libindy, settings};
+        use aries_vcx::handlers::connection::connection::Connection;
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:51]: `libindy`, `Connection` seem unused (maybe others)

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            "read_host": "localhost",
+            "write_host": "localhost",
+            "port": 3306,
+            "db_name": db_name,
+            "default_connection_limit": 50
+          }).to_string();
+            let storage_credentials = json!({
+            "user": "root",
+            "pass": "mysecretpassword"
+        }).to_string();
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:76]: incorrect allignment

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#[macro_use]
+extern crate log;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:5]: Not used

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    #[cfg(feature = "mysql_test")]
+    pub async fn setup_mysql_walletdb() -> Result<String, sqlx::Error> {
+        info!("Running query.");
+        let db_name = format!("mysqltest_{}", uuid::Uuid::new_v4().to_string()).replace("-", "_");
+        let url = "mysql://root:mysecretpassword@localhost:3306";
+        let mut connection = MySqlConnection::connect(url).await?;
+        let query = format!("CREATE DATABASE {};", db_name);
+        let query = sqlx::query(&query);
+        let res = query.execute(&mut connection).await;
+        warn!("Create database result: {:?}", res);
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:28]: Does not need to be `warn` level (I am not a fan of logging this at all)

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        connection.close().await;
+
+        let url = format!("mysql://root:mysecretpassword@localhost:3306/{}", db_name);
+        let mut connection = MySqlConnection::connect(&url).await?;
+        let res = sqlx::migrate!("./migrations")
+            .run(&mut connection)
+            .await;
+        warn!("Create tables result: {:?}", res);
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:36]: dtto

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+extern crate serde_json;
+#[cfg(feature = "mysql_test")]
+#[macro_use]
+extern crate sqlx;
+
+#[cfg(feature = "mysql_test")]
+mod dbtests {
+    use sqlx::{Connection, MySqlConnection};
+    use sqlx::Executor;
+    use sqlx::prelude;
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:16]: Not used (I am not a fan of the star notation, though)

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    use sqlx::prelude::*;
+
+    #[cfg(feature = "mysql_test")]
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:19]: This feature is configured in the outer module, so I would assume it is not necessary here (applies to the `test` module below as well)

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    pub async fn setup_mysql_walletdb() -> Result<String, sqlx::Error> {
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:20]: Elsewhere, we established a pattern where functions like this are put into a `test_utils` module (with a feature of the same name, which is [included](https://github.com/hyperledger/aries-vcx/pull/345/files#diff-3e6ca8175bf22b164caebe6ae1ce69a105be029e5c329cb15cfe15c260f7e73cR22) under `mysql_test` feature anyway)

File: aries_vcx/tests/test_mysql_wallet.rs [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+#[cfg(feature = "mysql_test")]
+#[macro_use]
+extern crate sqlx;
+
+#[cfg(feature = "mysql_test")]
+mod dbtests {
```
Comment (mirgee) [inline aries_vcx/tests/test_mysql_wallet.rs:13]: This outer module seems quite superfluous at the moment

File: agency_client/Cargo.toml [hunk 10]
```diff
@@ -20,30 +20,10 @@ reqwest = "0.9.5"
 regex = "1.1.0"
 rmp-serde = "0.13.7"
 rust-base58 = "0.0.4"
-indy = "1.16.0-post-59"
+indy = { git = "https://gitlab.com/evernym/verity/vdr-tools.git", rev = "59e1ecc21c9a067" }
```
Comment (mirgee) [inline agency_client/Cargo.toml:23]: It's a pity the gitlabl repo is not versioned using tags (git revisions can't be ordered / compared easily)

**Review by mirgee — CHANGES_REQUESTED** [review]
just a few observations

File: agency_client/Cargo.toml (hunk 10 above)
Comment (Patrik-Stas) [reply agency_client/Cargo.toml:23]: True, I believe they will start version it eventually though.

**Comment by Patrik-Stas** [comment]
- closed this in favor of new PR https://github.com/hyperledger/aries-vcx/pull/385

**Pull request closed without merging**